test/utils/readings.go

```diff
@@ -0,0 +1,66 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+)
+
+// TestDeviceVirtualReading waits for device-virtual to produce readings by querying core-data
+// up to a maximun number
+func TestDeviceVirtualReading(t *testing.T) {
+	t.Run("query readings", func(t *testing.T) {
+		var count int
+
+		// wait device-virtual producing readings with maximum 60 seconds
+		for i := 1; ; i++ {
+			time.Sleep(1 * time.Second)
+			req, err := http.NewRequest("GET", "http://localhost:59880/api/v2/event/count", nil)
+			if err != nil {
+				fmt.Print(err)
+				return
+			}
+
+			resp, err := http.DefaultClient.Do(req)
+			if err != nil {
+				fmt.Print(err)
+				return
+			}
+			defer resp.Body.Close()
+
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				fmt.Print(err)
+				return
+			}
+
+			mapContainer := make(map[string]json.RawMessage)
+			err = json.Unmarshal(body, &mapContainer)
+			if err != nil {
+				fmt.Print(err)
+				return
+			}
+
+			c := mapContainer["Count"]
+			count, _ = strconv.Atoi(string(c))
```

Comment:
This is error prone. Count may not exist and may not be convertible to int.

An inline struct can be defined to unmarshal the body directly to the right data type and produce errors otherwise.